irods/library/irods_move

```diff
@@ -1,142 +1,175 @@
-
-  if [ -z "$entityId" ]
-  then
-    local parentColl
-    parentColl=$(dirname "$entity")
-
-    local dataName
-    dataName=$(basename "$entity")
-
-    entityId=$(quest "select DATA_ID where COLL_NAME = '$parentColl' and DATA_NAME = '$dataName'")
-  fi
-
-  printf '%s' "$entityId"
-}
-
-
-quest()
-{
-  local query="$*"
-
-  local response
-  if ! response=$(iquest --no-page '%s' "$query" 2>&1)
-  then
-    fail "$response"
-  fi
-
-  if ! [[ "$response" =~ ^CAT_NO_ROWS_FOUND ]]
-  then
-    printf '%s' "$response"
-  fi
-}
-
-
-require()
-{
-  local var="$1"
-
-  if [ -z "${!var}" ]
-  then
-    fail "variable '$var' must be defined"
-  fi
-}
-
-
-fail()
-{
-  local msg="$*"
-
-  # shellcheck disable=SC2016
-  jq --compact-output --monochrome-output --null-input --arg msg "$msg" \
-     '{failed: true, msg: $msg}' \
-    >&2
-
-  exit 1
-}
-
-
-succeed()
-{
-  local changed="$1"
-
-  # shellcheck disable=SC2016
-  jq --compact-output --monochrome-output --null-input --arg changed "$changed" \
-     '{changed: ($changed == "true")}' \
-    >&2
-
-  exit 0
-}
-
-
-set -e
-main "$@" 2>&1
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os
+import ssl
+from ansible.module_utils.basic import AnsibleModule
+from irods.session import iRODSSession
+
+
+DOCUMENTATION = r'''
+---
+module: irods_move
+
+short_description: Rename data collection/object
+
+# If this is part of a collection, you need to use semantic versioning,
+# i.e. the version is of the form "2.5.0" and not "2.4".
+version_added: "1.0.0"
+
+description: An ansible module for renaming a data object or collection. To ensure
+idempotency, if the source doesn't exist, but the destination does, it is
+assumed that the move already happened and the module succeeds.
+
+options:
+    source:
+        description: the current absolute path to the collection or data object
+        type: str
+    destination:
+        description: the new absolute path to the collection or data object
+        type: str
+
+# Specify this value according to your collection
+# in format of namespace.collection.doc_fragment_name
+extends_documentation_fragment:
+    - my_namespace.my_collection.my_doc_fragment_name
+
+author:
+    - Fenn Garnett (@Fennersteel)
+'''
+
+EXAMPLES = '''
+# Move iRODS data object/collection
+- name: move data object/collection
+    irods_move:
+        source: /testing/home/rods/test_src
+        destination: /testing/home/rods/test_dst
+        host: "{{ groups['irods_catalog'][0] }}"
+        port: 1247
+        admin_user: rods
+        admin_password: rods
+        zone: testing
+'''
+
+RETURN = '''
+exc:
+    description:
+        - type of last iRODS exception thrown
+        - empty string when none is thrown
+    type: str
+    returned: always
+exc_msg:
+    description:
+        - message of the last iRODS exception thrown
+        - empty string when none is thrown
+    type: str
+    returned: always
+'''
+
+try:
```

Comment:
Since the python-irodsclient modules are imported outside of this try-except clause, it's not needed.  `USE_IRODS_CLIENT` will always be set to `True`.  Either move the python-irodsclient module import inside the try block, or remove the try-except clause and usage of the variable `USE_IRODS_CLIENT`.